main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala

```diff
@@ -17,6 +17,72 @@ package ca.uwaterloo.flix.api.lsp.provider.completion
 
 import ca.uwaterloo.flix.language.ast.Symbol
 
+/**
+  * Represents changes between ASTs.
+  */
+object DeltaContext {
+  /**
+    * Merges two DeltaContext.
+    *
+    * Removes duplicates that has an older timestamp, and deltas older than 5 minutes.
+    *
+    * Limits deltas to 1000 elements.
+    *
+    * Sorts the list with the most recent delta as first element.
+    *
+    * @param d1  the first DeltaContext.
+    * @param d2  the second DeltaContext.
+    * @return    a DeltaContext with the of changes as List[Delta],
+    *            where the most recent is the first element of the list.
+    */
+  def mergeDeltas(d1: DeltaContext, d2: DeltaContext): DeltaContext = {
+    val newDeltas = {
+      if (d2.deltas.isEmpty) {
```

Comment:
This special case is not really needed. The code will still work without it, so I would remove it for clarity.